chainstate/src/detail/error.rs

```diff
@@ -48,6 +49,20 @@ pub enum BlockError {
     BestBlockLoadError(PropertyQueryError),
     #[error("Failed to load block")]
     BlockLoadError(PropertyQueryError),
+    // FIXME: I know that this kind of error was frowned upon in the past, but is it that bad?
```

Comment:
The problem isn't the details in PropertyQueryError. The problem is the inherent meaning. As I explained in another PR to a colleague, if we, for example, request a block and we don't find it, it could mean many different things depending on the context. In one context it might be OK because we're just checking. In another context it might be catastrophic as it's a previous block of a block we already have. The problem is that PropertyQueryError doesn't convey the context. It will never know the context no matter what we do, because it's built to be a generic querying layer. It's like trying to put blockchain-specific errors in some SQL database errors layer. It'll never work. 

The right thing to do is to specify errors in the higher-level context relative to what they mean in that context, because PropertyQueryError cannot have all the information it needs to make a judgment in all cases.

Reply:
If enums are becoming big for some context, feel free to create sub-enums. That's OK.